src/main/scala/regmapper/Annotation.scala

```diff
@@ -2,22 +2,196 @@
+}
+
+/**
+  * Chisel Annotation
+  *
+  * This creates the firrtl annotation
+  *
+  * @param target
+  * @param regMappingSer
+  */
+case class RegMappingChiselAnnotation(
+  target: RawModule,
+  regMappingSer: RegistersSer) extends ChiselAnnotation with RunFirrtlTransform {
+  def toFirrtl: RegFieldDescMappingAnnotation = RegFieldDescMappingAnnotation(target.toNamed, regMappingSer)
+
+  def transformClass: Class[RegMappingDumpTransform] = classOf[RegMappingDumpTransform]
+}
+
+class RegMappingDumpTransform extends Transform {
+  def inputForm: CircuitForm = LowForm
+
+  def outputForm: CircuitForm = LowForm
+
+  def execute(state: CircuitState): CircuitState = {
+    state.annotations.foreach {
+      case RegFieldDescMappingAnnotation(t, desc) => println(desc)
     }
+    state
+  }
+}
+
+object GenRegDescsAnno {
+  // Each class which has regmaps has instance counters
+  private var instanceCounters = scala.collection.mutable.Map[String, Int]()
+
+  def addInstanceCounter(key: String) : Unit = {
+    instanceCounters +=  ( key -> 0)
+  }
+
+  def getInstanceCount(name: String, baseAddress: BigInt) : Int = {
+    val nameAddress = s"${name}.${baseAddress}"
+
+    if (! (instanceCounters isDefinedAt nameAddress)) {
+      addInstanceCounter(nameAddress)
+    }
+    val cnt = instanceCounters(nameAddress)
+    val newCnt = cnt + 1
+    instanceCounters(nameAddress) = newCnt
+    cnt
+  }
+
+  def makeRegMappingSer(rawModule: RawModule,
+    moduleName: String,
+    baseAddress: BigInt,
+    width: Int,
+    byteOffset: Int,
+    bitOffset: Int,
+    regField: RegField): RegFieldSer = {
+
+    val anonRegFieldName = s"unnamedRegField${byteOffset.toHexString}_${bitOffset}"
+    val regFieldName = regField.desc.map {
+      _.name
+    }.getOrElse("")
+    val selectedRegFieldName = if (regFieldName.isEmpty /* selectedName.isEmpty */) anonRegFieldName else regFieldName
+    val map = Map[BigInt, (String, String)]() // TODO
+
+    val desc = regField.desc
+
+    val byteOffsetHex = s"0x${byteOffset.toInt.toHexString}"
+
+    val regFieldDescSer = RegFieldDescSer(
+      byteOffset = byteOffsetHex,
+      bitOffset = bitOffset,
+      bitWidth = width,
+      name = selectedRegFieldName,
+      desc = desc.map {
+        _.desc
+      }.getOrElse("None"),
+      group = desc.map {
+        _.group.getOrElse("None")
+      }.getOrElse("None"),
+      groupDesc = desc.map {
+        _.groupDesc.getOrElse("None")
+      }.getOrElse("None"),
+      accessType = desc.map {
+        _.access.toString
+      }.getOrElse("r"), // TODO default?
+      wrType = desc.map(_.wrType.toString).getOrElse("None"),
+      rdAction = desc.map(_.rdAction.toString).getOrElse("None"),
+      volatile = desc.map(_.volatile).getOrElse(false),
+      hasReset = desc.map { d => if (d.reset != None) true else false }.getOrElse(false), // TODO ugly
+      resetValue = BigInt(0), // TODO desc.map{_.reset}.getOrElse(BigInt(0))
+      enumerations = map
```

Comment:
Not sure what the enumerations does

Reply:
Enumerations is a placeholder for future enumerations